### PR TITLE
Fix GH-41: Improve calendar import logic and ID mapping when creating new calendars

### LINKED ISSUE #41: [BUG] Migrating from SC | Can't import calendar notes
**Describe the bug**
When trying to import a calendar that originates from the old SC, the calendar itself comes in fine, but notes are not imported.

In the console, there's a warning that says: `No calendar found for import into: default_temp`

**To Reproduce**
On a fresh install of SCR on Foundry 14
1. Go to Configure -> Import/Export
2. Select file: [simple-calendar-export-doodletown-formatted.json](https://github.com/user-attachments/files/27305767/simple-calendar-export-doodletown-formatted.json)
3. Accept all defaults (all the default checkbox and dropdown values) and click Import
4. Close all SCR windows and re-open to see the calendar changes. The calendar is imported, but the notes are not there. Console warning shows `No calendar found for import into: default_temp`

**Expected behavior**
Notes from my configuration .json file should import correctly into the calendar I'm also importing at the same time.

**Screenshots**

Image

**Foundry Information**
 - Foundry Version: 14 v2
 - Game System: DnD5e
 - Simple Calendar Version: 2.6.0

**Browser (please complete the following information):**
 - Browser: Chrome/Vivaldi

**Additional context**
I've cloned the repository and built it in order to see if I could figure this out a bit better. Specifically, I found this in the `importCalendarSave()` method:

https://github.com/Fireblight-Studios/foundryvtt-simple-calendar/blob/40ee5409c23f315f2746f1c7d5f17a06e04e11d6/src/classes/applications/configuration-app.ts#L1859

The value `importInto` (set above this line) gets set to `"default_temp"`, and the calendar in `this.calendars` has the ID `"default_temp"`. So, if I change the above line to not remove the `_temp` part:

```ts
return c.id === importInto;
```

it then starts working, and my notes all get imported correctly.

~~Seems to me like the calendar's permanent saved name is `default_temp`, which maybe is a bug on its own?~~ (This might not be true since exporting the calendar gives its ID as "default") Still though, this logic in this part of the application being aware of `_temp` IDs seems like asking for trouble, since I think that knowledge is supposed to live in `calendar-manager.ts`, or maybe somewhere new (I haven't familiarized myself a ton with the codebase yet, so I don't think I have the right answer). Either way, if it were me, I would recommend centralizing that concern somewhere that isn't here. 😅 

(Fully appreciate though that you inherited this codebase and didn't completely write it yourself 👍 )

### PR DESCRIPTION
**Pull request by Arctis-Fireblight**
Fixes #41 
- Added a lookup table to map old and new calendar IDs during imports.
- Handled cases where calendar IDs include "_temp" and normalized them.
- Ensured creation of new calendars and proper mapping of notes if "new" is selected during import.
- Bumped version to v2.6.1 for an immediate release after merge.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved calendar import to correctly target notes when importing into new calendars.
  * Fixed temporary calendar identifier handling during import operations.

### CODE REVIEW

File: src/classes/applications/configuration-app.ts [hunk 1]
```diff
@@ -1850,11 +1856,16 @@ export default class ConfigurationApp extends FormApplication {
                         )
                     )
                         continue;
-                    const importInto = getTextInputValue(
+                    let importInto = getTextInputValue(
                         `.fsc-import-export .fsc-importing .fsc-file-details select[data-for-cal="${calId}"]`,
                         calId,
                         this.appWindow
-                    );
+                    ).replace("_temp", ""); // Sometimes will have "_temp" appended, need to normalize (GH-41)
```
[the remark this answers is not in the document]
Comment (Arctis-Fireblight) [reply src/classes/applications/configuration-app.ts:1863]: I havent seen an issue in testing? My understanding of the code in question is that it looks for a dropdown that matches the notes id for a corresponding calendar, and falls back to the notes id if one is not found. The imported json may or may not contain calendars at all let alone ones that match the given notes entry.
Originally import logic would only import notes for calendars contained in the json backup file, but the UI allows for users to backup and restore notes independently, so this logic was split apart in the importer a few commits back.

Though this split caused a few regressions that are addressed by this PR. Notably that if the backup contains calendars their id's will have _temp appended to them, and that if "new calendar" is selected, notes could not be imported at all.

[the remark this answers is not in the document]
Comment (Arctis-Fireblight) [reply src/classes/applications/configuration-app.ts:1863]: An edge case is an edge case, and I have been burned by about half a dozen of them in this method alone since I took over from the previous maintainer. Let me try your suggested change in my local copy and test.
It would be weird for them to reassign notes only imports to different calendars, but the UI still technically allows for it... So I should probably handle this edge case.
